### PR TITLE
feat: Adding command to delete API / API version

### LINKED ISSUE #203: Please add swaggerhub api:delete
**Is your feature request related to a problem? Please describe.**
We often create seperate API's as part of CI / CD workflows for testing (not new versions of the same API as the version list gets very polluted)

As part of a cleanup job on branch delete we would like to use something like

```
if swaggerhub api:get ORG/my-api-${{ env.GITHUB_REF_SLUG_URL }}; then
  swaggerhub api:delete ORG/my-api-${{ env.GITHUB_REF_SLUG_URL }}
fi
```

**Describe the solution you'd like**
The same as described above 

**Describe alternatives you've considered**
Manual revision is not fun for anyone

### PR DESCRIPTION
**Pull request by barryadk**
Closes #203

### CODE REVIEW

**Review by ali-smartbear — APPROVED** [review]

**Review by conor-ward — DISMISSED** [review]

**Review by ben-smartbear — DISMISSED** [review]

File: src/commands/api/delete.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+DeleteAPICommand.args = [{ 
+  name: 'OWNER/API_NAME/[VERSION]',
+  required: true,
+  description: 'API to delete in SwaggerHub'
+}]
+
+DeleteAPICommand.flags = {
+  force: flags.boolean({
+    char: 'f', 
+    description: 'file location of API to create'
```
Comment (conor-ward) [inline src/commands/api/delete.js:58]: Not sure what to have here 'skip confirmation when deleting API'?

**Review by conor-ward — APPROVED** [review]

**Review by ben-smartbear — APPROVED** [review]

File: src/commands/api/delete.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+class DeleteAPICommand extends BaseCommand {
+
+  async run() {
+    const { args, flags } = this.parse(DeleteAPICommand)
+    const apiPath = getApiIdentifierArg(args)
+    const [owner, name, version] = splitPathParams(apiPath)
+
+    if (version) {
+      this.logCommandSuccess = this.setSuccessMessage('deletedApiVersion')
+    } else if (!flags.force && await this.confirmDeletion(name) == false) {
```
Comment (ponelat) [inline src/commands/api/delete.js:16]: We're testing explicitly for `false` here, which means undefined/null/etc will still pass as okay. Best practice is to be more inclusive and test for falsey-ness.
A possible edge case: What if the user hits Cmd/Ctrl-C during this prompt, will the command execute or bail out?

Comment (conor-ward) [reply src/commands/api/delete.js:16]: I just tested this, it bails out. I think its because the default response for `confirmDeletion` is `false`. So anything other than responding with `y` is `false`.

Comment (barryadk) [reply src/commands/api/delete.js:16]: @ponelat Would `!= true` or `!== true` be a better option? Do you have any recommendation for this type of the method return comparision?

Comment (ponelat) [reply src/commands/api/delete.js:16]: `!== true` would be better (given the await, wrapping in in brackets is kinda meh... ie: `&& !(await this.confirmDeletion(name)`)
The thinking is that if-and-only-if it's confirmed (ie: true), go and delete. If anything else, don't.
Thanks for testing @conor-ward .

### Side note on `==` vs `===`
In short, always use `===` (and `!==`).
The end :joy: 

- `==` will try coerce to same type, so `1 == '1' // true`
- `===` has two quirks, if you don't want them you'll need Object.is. `NaN === NaN // false. wtf` but `Object.is(NaN, NaN) // true`

**Review by conor-ward — APPROVED** [review]
